### ISSUE TITLE
Class weights for imbalanced topk classification

### ISSUE THREAD

**Issue by VictorCallejas** [oval-group/smooth-topk#9]
Hi there,

I was wondering if there is a way to apply class weights to the loss function?

I have used the topk loss with good results. Still I have a pretty imbalanced dataset and class weights could be very convenient.

Thanks.

**Comment by lberrada** [comment]
Hi, there are two possible approaches to correct the imbalance.

First you can change the importance of the different samples according to their class. For this  simply replace the average by a weighted average of your choice in topk/svm.py.

The other possibility is to use the hyper-parameter alpha, which plays the role of a margin in the loss function. Currently it is a scalar defined statically in the loss function, but you can modify the loss function to (1) use a vector per sample rather than a single scalar and (2) pass it as an argument of the forward pass. This would allow you to define a larger margin for examples that are deemed important (rare) in the dataset.
